Samples/iOS-Swift/iOS-Swift/Tools/UIAssert.swift

```diff
@@ -0,0 +1,95 @@
+        
+        while let presented = targetViewController.presentedViewController {
+            targetViewController = presented
+        }
+        
+        guard let targetView = targetViewController.view else { return }
+        
+        if view.superview != targetView {
+            view.removeFromSuperview()
+            
+            targetView.addSubview(view)
+            
+            let constraints = [
+                view.leftAnchor.constraint(equalTo: targetView.leftAnchor, constant: 0),
+                view.rightAnchor.constraint(equalTo: targetView.rightAnchor, constant: 0),
+                view.bottomAnchor.constraint(equalTo: targetView.bottomAnchor, constant: 0)
+            ]
+            NSLayoutConstraint.activate(constraints)
+        }
+    }
+    
+    func reset() {
+        isFailed = false
+    }
+    
+    static func isTrue(_ value: Bool, _ errorMessage: String? = nil) {
+        shared.assert(success: value, errorMessage: errorMessage)
+    }
+    
+    static func isFalse(_ value: Bool, _ errorMessage: String? = nil) {
+        shared.assert(success: !value, errorMessage: errorMessage)
+    }
+    
+    static func isEqual<T>(_ first: T, _ second: T, _ errorMessage: String? = nil)  where T: Equatable {
+        shared.assert(success: first == second, errorMessage: errorMessage)
+    }
+        
+    static func notNil(_ value: Any?, _ errorMessage: String? = nil) {
+        shared.assert(success: value != nil, errorMessage: errorMessage)
+    }
+    
+    static func isNil(_ value: Any?, _ errorMessage: String? = nil) {
+        shared.assert(success: value == nil, errorMessage: errorMessage)
+    }
+    
+    static func fail(_ errorMessage: String? = nil) {
+        shared.assert(success: false, errorMessage: errorMessage)
+    }
+    
+    static func hasViewControllerLifeCycle(_ span: Span, _ viewController: String) {
+        guard let children = span.children() else {
+            shared.assert(success: false, errorMessage: "\(viewController) span has no children")
+            return
+        }
+        
+        let loadViewSpan = children.first { $0.context.spanDescription == "loadView" }
+        let viewDidLoadSpan = children.first { $0.context.spanDescription == "viewDidLoad" }
+        let viewWillAppearSpan = children.first { $0.context.spanDescription == "viewWillAppear" }
+        let viewDidAppearSpan = children.first { $0.context.spanDescription == "viewDidAppear" }
+        let viewAppearingSpan = children.first { $0.context.spanDescription == "viewAppearing" }
+        
+        notNil(loadViewSpan, "\(viewController) has no loadView span")
+        notNil(viewDidLoadSpan, "\(viewController) has no viewDidLoad span")
+        notNil(viewWillAppearSpan, "\(viewController) has no viewWillAppear span")
+        notNil(viewDidAppearSpan, "\(viewController) has no viewDidAppear span")
+        notNil(viewAppearingSpan, "\(viewController) has no viewAppearing span")
```

Comment:
`l`: There is a lot of duplicate code in here. Maybe something like this would make it a bit cleaner.
```suggestion
        func hasChildren(spanDescriptions: [String]) {
            spanDescriptions.forEach { spanDescription in
                let span = children.first { $0.context.spanDescription == spanDescription }
                notNil(span, "\(viewController) has no \(spanDescription) span")
            }
        }
        
        hasChildren(spanDescriptions: [
            "loadView", "viewDidLoad", "viewWillAppear", "viewDidAppear", "viewAppearing"
        ])    
```